src/main/java/com/metamx/common/parsers/TimestampParser.java

```diff
@@ -28,76 +28,107 @@
 {
   private static final Logger log = new Logger(TimestampParser.class);
 
-  public static Function<String, DateTime> createTimestampParser(
+  public static Function<Object, DateTime> createTimestampParserGeneric(
```

Comment:
i tried doing that but its not same for all the cases, nano and ruby have different logic since the Number cannot be directly parsed to timestamp.